File: libs/typeorm/src/entities/IgnoredUser.ts
import { Column, Entity, Index, PrimaryColumn } from 'typeorm';

@Entity('users_ignored', { schema: 'public' })
export class IgnoredUser {
  @PrimaryColumn('text', { primary: true, name: 'id' })
  id: string;

  @Index()
  @Column('text', { name: 'login', nullable: true })
  login: string | null;

  @Index()
  @Column('text', { name: 'displayName', nullable: true })
  displayName: string | null;
}

File: libs/gomodels/ignored_user.go
package model

import (
	"github.com/guregu/null"
)

type IgnoredUser struct {
	ID          string      `gorm:"primary_key;AUTO_INCREMENT;column:id;type:TEXT;" json:"id"`
	Login       null.String `gorm:"column:login;type:TEXT;"                         json:"login"`
	DisplayName null.String `gorm:"column:displayName;type:TEXT;"                   json:"displayName"`
	Force       bool        `gorm:"column:force;type:BOOLEAN;default:false;"        json:"force"`
}

func (i *IgnoredUser) TableName() string {
	return "users_ignored"
}
